src/dune_rules/pkg_rules.ml

```diff
@@ -387,9 +387,27 @@ module Substitute = struct
     Action.Extension (module M)
 end
 
+type sys_poll =
+  { os_version : string option Memo.Lazy.t
+  ; os_distribution : string option Memo.Lazy.t
+  ; os_family : string option Memo.Lazy.t
+  }
```

Comment:
I would move this record to `Sys_poll` and then have a single function there to generate it . That way we can be sure that we aren't running the same thing twice.

Reply:
Ok, I added it there. I am a bit hesitant about having global state there, but we can always revisit this later.